### DISCUSSION TITLE
I have a question about shortcut which could work in WindiCSS but couldn't work in UNOCSS

### DISCUSSION THREAD

**Discussion by JohnYang02182** [unocss/unocss#2290] [Q&A] [answered]
I have tried to add the shortcut to define color parameters and I just try the way from https://windicss.org/features/shortcuts.html

```
shortcuts: {
    'btn': {
      'color': 'white',
      '&:hover': {
        'color': 'black',
      },
    },
  },
```
and this code could work in WindiCSS, but When I tried to run it in UNOCSS, the UNOCSS just return me type error. Do I miss something? 
If somebody could tell me, I would appreciate it very much.

**Comment by zyyv** [comment]
You can refer to shortcuts types.

https://github.com/unocss/unocss/blob/4959845937841a21e324a4beb2fb96a84f97a751/packages/core/src/types.ts#L182-L187

[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBADgtAJgAwD4BQUrAEYFcAu%2BA9gHZTawAecALLAJ6JQBOI5%2BJK%2BApgM77AA9HkKl0Q8BHRogA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq4cKgAtoMJRyoAuR8%2BcWGCwfeRYiAFpKZDh3ADdUKAjwiJhUQxMNZzESJ3JKGhhfOABtfkLhCDsASj4C2gYmVg4ubhqAXVyxavUgA&css=Q&options=N4XyA)

**Reply by JohnYang02182** [reply]
Thanks for your prompt reply, and with your help I have solved the type error issue, but the code:
``
"shortcuts: {
    'btn': {
      'color': 'white',
      '&:hover': {
        'color': 'black',
      },
    },
},"
``
[playground](https://uno.antfu.me/play/?html=DwEwlgbgBADgtAJgAwD4BQUrAEYFcAu%2BA9gHZTawAecALLAJ6JQBOI5%2BJK%2BApgM77AA9HkKl0mHAWJkYzMAFsAhs0bYOKRZs1CR09EPAR0aIA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq4cKgAtoMJRyoAuR8%2Bc5FhgsOR9EJwDnJQgTaHC5SmQ5EkiolkMlAGsiaUVkfDioBKSNKLFUqLkKUEMobgBaYNDw-yi4GKKS1GTK9ozs3Ih8wvj5FhNMrLKAsUiKyP4aGF84AG0loRE7AEo%2BSmWGJlYOLm5dgF1UsR31IA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)
still couldn't work in UNOCSS, and I figured out that UNOCSS haven't supported CSS-in-JS, I think I should find another way to define the color parameters.

**Reply by zyyv** [reply]
As this case, i would suggest you use rules. [Playground](https://uno.antfu.me/play/?html=DwEwlgbgBADgtAJgAwD4BQUrAEYFcAu%2BA9gHZTawAecALLAJ6JQBOI5%2BJK%2BApgM77AA9HkKl0mHAWJkYzMAFsAhs0bYOKRZs1CR09EPAR0aIA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq4cKKeoAuOAG0nz73JYYLDkyRz8-JQgTaE85SmQQ33CWQyUAayJpRWR8aKhY%2BI1wsQBdEiT-ClBDKG4AWkDg0IrnSLyC1ATy8OcU9MyIbNyY%2BRYTVLSiv1LfMt9%2BGhgqTy8FoRE7AEo%2BSkWGJlYOLm4tubFN9SA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

And this is a `staticRule` case, you can write some reg to use `dynamicRule`, it's up to you.

Refer to [custom rule docs](https://github.com/unocss/unocss#custom-rules).

**Comment by antfu** [comment, marked as answer]
It's how it been designed. UnoCSS is not a 1:1 WindiCSS placement, so we don't aim for feature parity. UnoCSS has a more powerful but different shortcut syntax.
